aspnetcore/fundamentals/servers/httpsys/samples/8.x/SampleApp/Program.cs

```diff
@@ -0,0 +1,57 @@
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.HttpSys;
+
+// <snippet_1>
+var builder = WebApplication.CreateBuilder(args);
+
+builder.WebHost.UseHttpSys(options =>
+{
+    options.AllowSynchronousIO = false;
+    options.Authentication.Schemes = AuthenticationSchemes.None;
+    options.Authentication.AllowAnonymous = true;
+    options.MaxConnections = null;
+    options.MaxRequestBodySize = 30_000_000;
+    options.UrlPrefixes.Add("http://localhost:5005");
+});
+
+builder.Services.AddRazorPages();
+// </snippet_1>
+
+// <snippet_2>
+var app = builder.Build();
+
+app.Use((context, next) =>
+{
+    var maxRequestBodySizeFeature = context.Features.GetRequiredFeature<IHttpMaxRequestBodySizeFeature>();
```

Comment:
The 3.x sample was using `Get<TFeature>` here instead and wasn't taking nullability into account. I updated these to use `GetRequiredFeature` instead simply to avoid needing null checks in each place where we get request or server features.